From 6f128a72bc0173690246a7468821628656340e39 Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob-chen@iotwrt.com>
Date: Wed, 7 Jun 2017 10:05:10 +0800
Subject: [PATCH 4/6] DEBIAN-ONLY: rockchipmpp: always use simple io mode

mpp have two io modes: simple and advance.
At current, we only support simple mode.

Signed-off-by: Jacob Chen <jacob-chen@iotwrt.com>
---
 gst/rockchipmpp/gstmppvideodec.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gst/rockchipmpp/gstmppvideodec.c b/gst/rockchipmpp/gstmppvideodec.c
index 7ec4efa..315dddf 100644
--- a/gst/rockchipmpp/gstmppvideodec.c
+++ b/gst/rockchipmpp/gstmppvideodec.c
@@ -185,9 +185,17 @@ static gboolean
 gst_mpp_video_dec_open (GstVideoDecoder * decoder)
 {
   GstMppVideoDec *self = GST_MPP_VIDEO_DEC (decoder);
+
+  gint io_mode = MPP_IO_MODE_SIMPLE;
+
+  mpp_create (&self->mpp_ctx, &self->mpi);
+
   if (mpp_create (&self->mpp_ctx, &self->mpi))
     return FALSE;
 
+  self->mpi->control (self->mpp_ctx, MPP_SET_TRANSACTION_MODE,
+      (gpointer) & io_mode);
+
   GST_DEBUG_OBJECT (self, "created mpp context %p", self->mpp_ctx);
   return TRUE;
 }
-- 
2.15.1

